File: FE/src/types/Post.ts
export type PostStatus =
  | "JUST_POSTED"
  | "FIRST_REACTION"
  | "EXPIRED";

export type Post = {
  id: string;

  author: string;
  
  authorId:number;

  title: string;
  text: string;

  image?: string;

  createdAt: string;

  status: PostStatus;

  likes: number;
  dislikes: number;

  score: number;

  tags?: string[];
};

File: BE/src/main/java/org/example/proiectps/entity/Post.java
package org.example.proiectps.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import org.example.proiectps.enums.PostStatus;


@Entity
@Table (name = "posts")
@Data
@NoArgsConstructor
@AllArgsConstructor


public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long postId;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User author;

    @Column(nullable = false)
    private String title;
    @Column(nullable = false)
    private String text;
    @Column(nullable = false)
    private LocalDateTime date;
    private String image;
    private PostStatus status=PostStatus.JUST_POSTED;

    ///TAGS
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "post_tags",
            joinColumns = @JoinColumn(name = "post_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    @JsonIgnore
    private List<Tag> tags = new ArrayList<>();
}